hs_core/views/__init__.py

```diff
@@ -653,6 +767,221 @@ def create_resource(request, *args, **kwargs):
 
 
 @login_required
+def create_user_group(request, *args, **kwargs):
+    group_form = GroupForm(request.POST, request.FILES)
+    if group_form.is_valid():
+        try:
+            new_group = group_form.save(request)
+            messages.success(request, "Group creation was successful.")
+            return HttpResponseRedirect(reverse('group', args=[new_group.id]))
+        except IntegrityError as ex:
+            if group_form.cleaned_data['name'] in ex.message:
+                message = "Group name '{}' already exists".format(group_form.cleaned_data['name'])
+                messages.error(request, "Group creation errors: {}.".format(message))
+            else:
+                messages.error(request, "Group creation errors:{}.".format(ex.message))
```

Comment:
Since the if condition just checks whether exception message includes the group name or not, I think it is best to append the exact exception message to group not exist custom message even if it is not user-friendly, perhaps something like "Group name already exists - exception message: ..." something like that. Just for your reference.